File: app/src/main/java/com/lifedawn/bestweather/theme/AppTheme.java
package com.lifedawn.bestweather.theme;

import android.content.Context;
import android.graphics.Color;
import android.util.TypedValue;

import com.lifedawn.bestweather.R;
import com.lifedawn.bestweather.weathers.FragmentType;

public class AppTheme {
	public static int getColor(Context context, int id) {
		TypedValue value = new TypedValue();
		context.getTheme().resolveAttribute(id, value, true);
		return value.data;
	}

	public static int getTextColor(FragmentType fragmentType) {

		switch (fragmentType) {
			case Detail:
			case Comparison:
				return Color.BLACK;
			default:
				return Color.WHITE;
		}

	}
}

File: app/src/main/java/com/lifedawn/bestweather/ui/theme/AppTheme.kt
package com.lifedawn.bestweather.ui.theme

import android.content.Context
import android.graphics.Color
import android.util.TypedValue
import com.lifedawn.bestweather.ui.weathers.enums.WeatherDataType

object AppTheme {
    fun getColor(context: Context, id: Int): Int {
        val value = TypedValue()
        context.theme.resolveAttribute(id, value, true)
        return value.data
    }

    @JvmStatic
    fun getTextColor(weatherDataType: WeatherDataType?): Int {
        return when (weatherDataType) {
            WeatherDataType.Detail, WeatherDataType.Comparison -> Color.BLACK
            else -> Color.WHITE
        }
    }
}